### PR TITLE
Fix TextSpan's operator==

### PR DESCRIPTION
**Pull request by abarth**
We forgot to compare the lengths of the lists.

### CODE REVIEW

**Comment by abarth** [comment]
@Hixie @pylaligand

**Comment by pylaligand** [comment]
LGTM

Thanks for the quick fix!

File: packages/flutter/test/painting/text_span_test.dart [hunk 1]
```diff
@@ -0,0 +1,30 @@
+// Copyright 2015 The Chromium Authors. All rights reserved.
```
Comment (pylaligand) [inline packages/flutter/test/painting/text_span_test.dart]: 2016

**Comment by Hixie** [comment]
There's places where we do this kind of deep equality already. We should probably use the same solution for all of them.

**Comment by Hixie** [comment]
lgtm

**Comment by abarth** [comment]
I'm going to handle the other locations in a later patch.

**Pull request merged**